crates/cdk-mintd/src/config.rs

```diff
@@ -210,7 +211,7 @@ pub struct MintInfo {
 
 impl Settings {
     #[must_use]
-    pub fn new(config_file_name: &Option<PathBuf>) -> Self {
+    pub fn new<P: Into<PathBuf>>(config_file_name: Option<P>) -> Self {
```

Comment:
NIT code style should use where https://github.com/cashubtc/cdk/blob/main/CODE_STYLE.md#generics.

This applies in a few places but ill just comment here. Won't block merging for it